content/news/2022-01-06-bevy-0.6/index.md

```diff
@@ -0,0 +1,1007 @@
+
+```rust
+#ifdef TEXTURE
+[[group(1), binding(0)]]
+var sprite_texture: texture_2d<f32>;
+#endif
+```
+
+This pattern is very useful when defining complicated / configurable shaders (such as Bevy's PBR shader).
+
+### Shader Imports
+
+<div class="release-feature-authors">authors: @cart</div>
+
+The new preprocessor supports importing other shader files (which pulls in their entire contents). This comes in two forms:
+
+Asset path imports:
+
+```rust
+# import "shaders/cool_function.wgsl"
+
+[[stage(fragment)]]
+fn fragment(input: VertexOutput) -> [[location(0)]] vec4<f32> {
+    return cool_function();
+}
+```
+
+Plugin-provided imports, which can be registered by Bevy Plugins with arbitrary paths:
+
+```rust
+# import bevy_pbr::mesh_view_bind_group
+
+[[stage(vertex)]]
+fn vertex(vertex: Vertex) -> VertexOutput {
+    let world_position = vec4<f32>(vertex.position, 1.0);
+    var out: VertexOutput;
+    // The `view` variable comes from the imported bind group 
+    out.clip_position = view.view_proj * world_position;
+    return out;
+}
+```
+
+We also plan to experiment with using Naga for "partial imports" of specific, named symbols (ex: import a specific function or struct from a file).
+
+### Pipeline Specialization
+
+<div class="release-feature-authors">authors: @cart</div>
+
+When shaders use a preprocessor and have multiple permutations, the associated "render pipeline" needs to be updated to accommodate those permutations (ex: different Vertex Attributes, Bind Groups, etc). To make this process straightforward, we added the SpecializedPipeline trait, which allows defining specializations for a given key:
+
+```rust
+impl SpecializedPipeline for MyPipeline {
+    type Key = MyPipelineKey;
+    fn specialize(&self, key: Self::Key) -> RenderPipelineDescriptor {
+        // define a specialized pipeline for the given key here
+    }
+}
+```
+
+Implementors of this trait can then easily and cheaply access specialized pipeline variants (with automatic per-key caching and hot-reloading). If this feels too abstract / advanced, don't worry! This is a "mid level power-user tool", not something most Bevy App developers need to contend with.
+
+### Simpler Shader Stack
+
+Bevy now uses [Naga](https://github.com/gfx-rs/naga) for all of its shader needs. As a result, we were able to remove all of our complicated non-rust shader dependencies: `glsl_to_spriv`, `shaderc`, and `spirv_reflect`. `glsl_to_spriv` was a major producer of platform-specific build dependencies and bugs, so this is a huge win!
+
+### Features Ported to the New Renderer
+
+Render logic for internal Bevy crates had to be rewritten in a number of cases to take advantage of the new renderer. The following people helped with this effort:
+
+* bevy_sprites: @cart, @StarArawn 
+* bevy_pbr: Rob Swain (@superdump), @aevyrie, @cart, @zicklag, @jakobhellermann
+* bevy_ui: @Davier
+* bevy_text: @Davier
+* bevy_gltf: Rob Swain (@superdump)
```

Comment:
if [#bevyengine/3460](https://github.com/bevyengine/bevy/pull/3460) makes it into the release, i think adding a bullet point for 2D meshes or adding @Davier to `bevy_sprites` would be great

Reply:
thanks for the heads up. ill update this post to accommodate new changes if/when they get merged